src/core/include/enum.h

```diff
@@ -48,6 +48,13 @@ enum class Status {
     Finished,  //! all is done
 };
 
+enum class GraphTool {
+    Default = -1,     // used in experiment designer, opens the 'light' inspector
+    Select = 0,     // Select mode
```

Comment:
this enum is related to the GUI only; it should not be in the `core`
you should move it to the place where it's actually used (basegraphgl)

I think that once the full inspector is open, we should never show the light inspector, so I guess that we could have only 3 modes here... (note that we could also make the full inspector available for the experiments)

BTW, in summary, those modes define the behaviour of a mouse click, right?
then, since none of them are actually "tools", wouldn't be better to define the enum as something like `SelectionMode`  instead of `GraphTool`? (Node and Edge could also be a bit more informative, e.g. NodeEdit and EdgeEdit)